src/main/scala/cromwell/engine/package.scala

```diff
@@ -34,4 +34,9 @@ package object engine {
     override def toString: String = "Succeeded"
     override val isTerminal = true
   }
+
+  trait WorkflowDescriptor {
+    def name: String
+    def id: WorkflowId
+  }
```

Comment:
This does collapse with my PR a little bit, because I implemented something similar except I *just* passed the WorkflowId around.  Yours is probably better, though I haven't fully read through all the code.  Don't worry I'll handle the rebase since my stuff is less important.

What about maybe including the WdlNamespace in that trait?

```scala
trait WorkflowDescriptor {
  def namespace: WdlNamespace
  val name = namespace.workflows.head.name
  def id: WorkflowID
}
```